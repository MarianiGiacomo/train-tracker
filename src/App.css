.vehicle-tracker-header {
    position: fixed;
    z-index: 3000;
    width: 100%;
}

.logo {
    width: 120px;
    height: 31px;
    background: rgba(255, 255, 255, 0.2);
    margin: 16px 24px 16px 0;
    float: left;
}

.vehicle-tracker-menu {
    line-height: 64px;
}

.vehicle-tracker-content-wrapper {
    padding: 0 50px; 
    margin-top: 64px;
}

.vehicle-tracker-content {
    background: #fff;
    padding: 24px;
    min-height: 90vh; 
}

.vehicle-tracker-footer {
    text-align: center; 
}