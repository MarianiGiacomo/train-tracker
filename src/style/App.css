
.content-wrapper {
    margin: 10px;
    background: #fff;
}