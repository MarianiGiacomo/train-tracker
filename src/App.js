import React, { Component } from 'react';
import { Layout, Menu } from 'antd';

import VehicleTrackerRest from './components/VehicleTrackerRest';
import VehicleTrackerWss from './components/VehicleTrackerWss';
import VehicleTrackerWsRest from './components/VehicleTrackerWsRest';

import './App.css';

const { Header, Content, Footer } = Layout;

class App extends Component {
  state = {
    selectedMenu: 'wss',
  };

  onMenuItemClick = (e) => {
    this.setState({
      selectedMenu: e.key,
    });
  }

  renderContent = () => {
    const { selectedMenu } = this.state;

    switch (selectedMenu) {
      case 'rest':
        return <VehicleTrackerRest />
      case 'wss': 
        return <VehicleTrackerWss />
      case 'wss-rest':
        return <VehicleTrackerWsRest />
      default: 
        return <VehicleTrackerWss />
    }
  }

  render = () => {
   
    return ( 
     <Layout>
      <Header className="vehicle-tracker-header">
        <div className="logo" />
        <Menu
          theme="dark"
          mode="horizontal"
          defaultSelectedKeys={['wss']}
          className="vehicle-tracker-menu"
          onClick={this.onMenuItemClick}
        >
          <Menu.Item key="wss">WebSocket</Menu.Item>
          <Menu.Item key="rest">REST</Menu.Item>
          <Menu.Item key="wss-rest">WS & REST</Menu.Item>
        </Menu>
      </Header>
      <Content className="vehicle-tracker-content-wrapper">
        <div className="vehicle-tracker-content">
          {this.renderContent()}
        </div>
      </Content>
      <Footer className="vehicle-tracker-footer">
        Vehicle tracker Â©2019
      </Footer>
    </Layout>
    );
  }
}

export default App;