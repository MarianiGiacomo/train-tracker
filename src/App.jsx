import React from 'react';
import { connect } from 'react-redux';

import AppHeader from './components/AppHeader';
import AppContent from './components/AppContent';

import './style/App.css';

import { getAndStoreMqttClient } from './reducers/mqttClientReducer';
import { storeTrainLocation } from './reducers/trainLocationReducer';
import { getAndStoreActiveTrainsByDate } from './reducers/trainsListReducer';
import helperFunctions from './utils/helperFunctions';

const App = (props) => {
  props.getAndStoreActiveTrainsByDate(new Date());
  props.getAndStoreMqttClient((message) => saveCoordinates(message));

  const saveCoordinates = async (message) => {
    const coordinates = helperFunctions.extractTrainLocation(message);
    props.storeTrainLocation(coordinates);
  };

  return (
    <div className="content-wrapper">
      <div className="header-component">
        <AppHeader title="Train Tracker" />
      </div>
      <div className="content-component">
        <AppContent />
      </div>

    </div>
  );
};

const mapDispatchToProps = {
  getAndStoreActiveTrainsByDate,
  getAndStoreMqttClient,
  storeTrainLocation,
};

export default connect(null, mapDispatchToProps)(App);
