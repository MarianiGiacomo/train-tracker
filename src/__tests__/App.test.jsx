import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';

import App from '../App'

afterEach(() => {
  cleanup();
  console.error.mockClear();
  console.warn.mockClear();
});

console.error = jest.fn();
console.warn = jest.fn();

describe('<App />', () => {
  test('Test snapshot', () => {
      const { container } = render(<App />);
      expect(container.firstChild).toMatchSnapshot();
  });

  test('Renders without errors or warnings', () => {
      render(<App />);
      expect(console.error).not.toHaveBeenCalled();
      expect(console.warn).not.toHaveBeenCalled();
  });

  test('Page renders the right components', () => {
      const { getByText } = render(<App />);
      expect(getByText('Train Tracker')).toBeTruthy();
  });
})


