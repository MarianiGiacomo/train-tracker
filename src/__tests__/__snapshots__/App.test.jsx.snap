// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<div
  style="width: 100%; height: 100%;"
/>
`;
